### PR TITLE
adds base code, cobra, kustomize deployment manifest

### PR DESCRIPTION
**Pull request by krishchow**
This PR adds the basic code for a COSI driver and is based on existing code from the [cosi-driver-minio](https://github.com/kubernetes-sigs/cosi-driver-minio). This includes:
- deployment manifests using kustomize
- Cobra to handle CLI setup, parsing and validation (similar to the other COSI projects)
- The base code (`main.go`, `cmd.go`), and the identity and provisioner servers.

This PR will act as the foundation for other PRs.

### CODE REVIEW

**Comment by krishchow** [comment]
It just adds the Makefile and logic around Cobra that was discussed in the comments around that PR. Additionally, I replicated the same structure for Kustomize as the other COSI repos.

File: cmd/sample-cosi-driver/main.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig := <-sigs
+		klog.InfoS("Signal received", "type", sig)
+		cancel()
+
+		<-time.After(30 * time.Second)
```
Comment (tparikh) [inline cmd/sample-cosi-driver/main.go:38]: What purpose does 30s wait before exiting serve?

File: pkg/provisioner.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+type ProvisionerServer struct {
+	provisioner string
+}
+
+// ProvisionerCreateBucket is an idempotent method for creating buckets
+// It is expected to create the same bucket given a bucketName and protocol
+// If the bucket already exists, then it MUST return codes.AlreadyExists
+// Return values
```
Comment (tparikh) [inline pkg/provisioner.go:31]: Might be worth clarifying the return value here is for `error`.

File: resources/ns.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: sample-cosi-driver
+  labels:
+    app.kubernetes.io/part-of: container-object-storage-interface
+    app.kubernetes.io/component: driver-sample
+    app.kubernetes.io/version: main
+    app.kubernetes.io/name: cosi-driver-sample
```
Comment (tparikh) [inline resources/ns.yaml:10]: Would recommend adding a new line to the file.

**Comment by tparikh** [comment]
Left few comments but PR looks good to me!

File: pkg/provisioner.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+//    nil -                   Bucket successfully created
+//    codes.AlreadyExists -   Bucket already exists. No more retries
+//    non-nil err -           Internal error                                [requeue'd with exponential backoff]
+func (s *ProvisionerServer) ProvisionerCreateBucket(ctx context.Context,
+	req *cosi.ProvisionerCreateBucketRequest) (*cosi.ProvisionerCreateBucketResponse, error) {
+
+	return nil, status.Error(codes.Unimplemented, "ProvisionerCreateBucket: not implemented")
```
Comment (tparikh) [inline pkg/provisioner.go:38]: Might we worth declaring error codes and messages else where as it is used in multiple places

File: cmd/sample-cosi-driver/main.go (hunk 1 above)
Comment (wlan0) [reply cmd/sample-cosi-driver/main.go:38]: it allows for the cancelled context to gracefully shutdown

**Review by wlan0 — APPROVED** [review]

**Pull request merged**